ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/LgrDB.hs

```diff
@@ -472,9 +473,12 @@ streamAPI immDB = StreamAPI streamAfter
       ImmDB.IteratorResult mblk -> (\blk -> NextBlock (blockRealPoint blk, blk)) <$> mblk
 
 {-------------------------------------------------------------------------------
-  Garbage collect points of previously applied blocks
+  Previously applied blocks
 -------------------------------------------------------------------------------}
 
+getPrevApplied :: IOLike m => LgrDB m blk -> STM m (Set (RealPoint blk))
+getPrevApplied LgrDB{..} = readTVar varPrevApplied
```

Comment:
Document and check the invariant that all blocks on the volatile part of the current chain (current fragment) are in `varPrevApplied`.